modules/compute/gke-node-pool/variables.tf

```diff
@@ -265,3 +265,31 @@ variable "service_account" {
     error_message = "service_account is deprecated and replaced with service_account_email and scopes."
   }
 }
+
+variable "reservation_type" {
+  description = "Type of reservation to consume"
+  type        = string
+  default     = null
+
+  validation {
+    condition     = contains(["NO_RESERVATION", "ANY_RESERVATION", "SPECIFIC_RESERVATION"], var.reservation_type)
+    error_message = "Accepted values are: {NO_RESERVATION, ANY_RESERVATION, SPECIFIC_RESERVATION}"
+  }
+}
+
+variable "specific_reservation" {
+  description = "Reservation resources to consume when targeting SPECIFIC_RESERVATION. Specify `compute.googleapis.com/reservation-name` as the key and the list of reservation names as the value."
+  type = object({
+    key    = string
+    values = list(string)
+  })
+  default = {
+    key    = null
+    values = null
+  }
+
+  validation {
+    condition     = (var.specific_reservation.key == "compute.googleapis.com/reservation-name" && length(var.specific_reservation.values) > 0) || (var.specific_reservation.key == null && var.specific_reservation.values == null)
```

Comment:
will check if the condition can be made more readable by spanning it over two lines.